### PR TITLE
Fixed Unexpected JSON Parse Errors

### PR DESCRIPTION
**Pull request by ppetrosh**
Fixed https://github.com/electricimp/Firebase/issues/20:

1. Handle split event notification messages arriving from the Firebase stream.
1. Unit tests added.

### CODE REVIEW

File: firebase.agent.nut [hunk 1]
```diff
@@ -594,16 +610,17 @@ class Firebase {
     // process the http response accordingly
     function _processResponse (request,callback) {
         request.sendasync(function(res) {
-            local data = res.body ;
+            local data = res.body;
             try {
-                data = http.jsondecode(data);
                 if (200 <= res.statuscode && res.statuscode < 300) {
-                    callback(null,data);
+                    data = http.jsondecode(data);
```
Comment (betzrhodes) [inline firebase.agent.nut]: Do we ever get a response from firebase where the status code is in the 200's but the response.body is an empty string? Do we want this situation to bubble up as a json parsing error?

**Comment by betzrhodes** [comment]
did the version number get updated in the readme & examples?

File: firebase.agent.nut (hunk 1 above)
Comment (ppetrosh) [reply firebase.agent.nut]: I have never seen this. And I'm afraid of changing anything without a use case to verify.

Comment (betzrhodes) [reply firebase.agent.nut]: I've seen it quite often in libraries and I noticed that Aron had a work around for this parsing error implemented in pull request #19 - https://github.com/electricimp/Firebase/pull/19 - though this pull request never got merged.  If this issue is not popping up then we might not need to handle it.

**Comment by ppetrosh** [comment]
Yep, the library version updated in the README. Thanks for catching this!

File: firebase.agent.nut (hunk 1 above)
Comment (ppetrosh) [reply firebase.agent.nut]: Add this check. Please review. I don't think an empty response should be considered as an error. Do you?

**Pull request closed without merging**